docs/source/java/vector.rst

```diff
@@ -0,0 +1,118 @@
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+===========
+ValueVector
+===========
+
+:class:`ValueVector` (which called Array in C++ implementation and the :doc:`the specification <../format/Columnar>`)
+is an abstraction that is used to store a sequence of values having the same type in an individual
+column. Internally, those values are represented by one or several buffers, the
+number and meaning of which depend on the vector’s data type.
+
+There are concrete subclasses of :class:`ValueVector` for each primitive data type
+and nested type described in the specification. There are a few differences in naming
+with the type names described in the specification:
+Table with non-intuitive names (BigInt = 64 bit integer, etc).
+
+It is important that vector is allocated before attempting to read or write,
+:class:`ValueVector` "should" strive to guarantee this order of operation:
+allocate > mutate > set valuecount > access > clear (or allocate to start the process over)
+
+Building ValueVector
+====================
+
+Note that the current implementation doesn't enforce the rule that Arrow objects are immutable.
+:class:`ValueVector` instances could be created directly by using new keyword, there are
+set/setSafe APIs and concrete subclasses of FieldWriter for populating values.
+
+For example, the code below shows how to build a :class:`BigIntVector`, in this case, we build a
+vector of the range 1 to 8 where the element that should hold the value 4 is nulled::
+
+   BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE);
+
+   BigIntVector vector = new BigIntVector("vector", allocator);
+   vector.allocateNew(8);
+   vector.set(0, 1);
+   vector.set(1, 2);
+   vector.set(2, 3);
+   vector.setNull(3);
+   vector.set(4, 5);
+   vector.set(5, 6);
+   vector.set(6, 7);
+   vector.set(7, 8);
+   vector.setValueCount(8); // this will finalizes the vector by convention.
+
+The :class:`BigIntVector` holds two ArrowBufs. The first buffer holds the null bitmap, which consists
+here of a single byte with the bits 0|0|0|0|1|0|0|0. The second buffer contains all the above values.
+As the fourth entry is null, the value at that position in the buffer is undefined. Note compared with
+set API, setSafe API would check value capacity before setting values and reallocate buffers if necessary.
+
+Here is how to build a vector using writer::
+
+   BigIntVector vector = new BigIntVector("vector", allocator);
+   BigIntWriter writer = new BigIntWriterImpl(vector);
+   writer.setPosition(0);
+   writer.writeBigInt(1);
+   writer.setPosition(1);
+   writer.writeBigInt(2);
+   writer.setPosition(2);
+   writer.writeBigInt(3);
```

Comment:
Also give an example for writing null?

Reply:
The fourth value is still null like the example with setAPI, I updated the comments.